### ISSUE TITLE
fancier support for customizable optimization

### ISSUE THREAD

**Issue by martinjankowiak** [uber/pyro#349]
- figure out if/how pytorch param groups interact with `PyroOptim`
- add necessary support for pytorch lr_scheduler
- make gradient clipping and the like cleaner

**Comment by eb8680** [comment]
I think the things mentioned have basically been addressed by `pyro.optim.multi` and `pyro.optim.lr_scheduler`, we can open more specific follow-up issues as necessary
